doc/DEVELOPMENT.rst

```diff
@@ -0,0 +1,266 @@
+
+
+Coding Style
+^^^^^^^^^^^^
+
+In the ``*.c`` files, tabs are used. In the ``meson.build`` and ``*.py`` files, spaces
+are used.
+
+To help you abide by this standard, you should install
+`EditorConfig <https://github.com/editorconfig>`__ for your preferred editor of
+choice, and load in the file found at ``.editorconfig`` in the root of the
+project.
+
+
+Platform-Specific Tips
+^^^^^^^^^^^^^^^^^^^^^^
+
+Windows
+"""""""
+
+TODO: complete this section.
+
+Linux (Debian/Ubuntu)
+"""""""""""""""""""""
+
+On an apt-based system (Debian/Ubuntu), ensure you have build dependencies
+installed:
+
+::
+
+    apt-get install meson cmake build-essential libsdl2-dev libsdl2-mixer-dev libogg-dev libopusfile-dev libpng-dev libzip-dev libx11-dev libwayland-dev
+
+
+macOS
+"""""
+
+Manual
+~~~~~
+
+On macOS, you need to begin with installing the Xcode Command Line Tools:
+
+::
+
+    xcode-select --install
+
+There are additional command line tools that you'll need. You can acquire those
+by using `Homebrew <https://brew.sh/>`__.
+
+Follow the instructions for installing Homebrew, and then install the following
+tools:
+
+::
+
+    brew install meson cmake pkg-config docutils imagemagick pygments
+
+
+The following dependencies are technically optional, and can be pulled in at
+build-time, but you're better off installing them yourself to reduce compile
+times:
+
+::
+
+    brew install freetype2 libzip opusfile libvorbis webp sdl2
+
+
+You'll also want to set the following environment variables on your shell's
+.zshrc or .bash_profile:
+
+::
+
+   export PKG_CONFIG_PATH="/usr/local/opt/openssl@1.1/lib/pkgconfig"
+
+
+As of 2020-02-18, you should **not** install the following packages via
+Homebrew, as the versions available do not compile against Taisei correctly.
+If you're having mysterious errors, ensure that they're not installed.
+
+* ``spirv-tools``
+* ``spirv-cross``
+* ``sdl2_mixer``
+
+Remove them with:
+
+::
+
+    brew remove spirv-tools spirv-cross sdl2_mixer
+
+
+Taisei-compatible versions will be pulled in at compile time.
+
+In addition, if you're trying to compile on an older version of macOS
+(e.x: <10.12), SDL2 may not compile correctly on Homebrew (as of 2019-02-19).
+Let ``meson`` pull in the corrected version for you via subprojects.
+
+
+Automatic
+~~~~~~~~~
+
+To do the above steps automatically, run ``scripts/macos-deps-install.sh``.
+
+Graphical Assets
+----------------
+
+Taisei's GFX library is made up of a collection of sprites, shaders, and a few
+3D models. The 3D models are almost exclusively used for background scenery
+(and a few other places, like the HUD), while the sprites are used in everything
+from UI elements, character portraits, to the
+danmaku bullets themselves.
+
+To modify the 3D models, you'll need `Blender <https://blender.org>`__, which is
+free and open source. Look for tutorials on YouTube for how to use it. The
+models themselves are located in ``resources/00-taisei.pkgdir/models``.
```

Comment:
Definitely worth documenting the export options for `.obj` models, since they are different from Blender's defaults.

![export options](https://0x0.st/iqCH.png)